src/main/java/org/eln2/mc/common/blocks/cell/WireCellBlock.kt

```diff
@@ -1,11 +1,209 @@
+        return Block.box(0.0, 0.0, 0.0, 16.0, 2.0, 16.0)
+    }
+
+    override fun onPlace(pState: BlockState, pLevel: Level, pPos: BlockPos, pOldState: BlockState, pIsMoving: Boolean) {
+
+
+
+        super.onPlace(pState, pLevel, pPos, pOldState, pIsMoving)
+
+
+        // get blocks around and call nechang
+
+        for (x in -1..1) {
+            for (z in -1..1) {
+                if (x == 0 && z == 0) continue
+                neChang(pLevel, pPos, BlockPos(pPos.x + x, pPos.y, pPos.z + z))
+            }
+        }
+
+
+
+
+
+    }
+
+    override fun createBlockStateDefinition(pBuilder: StateDefinition.Builder<Block, BlockState>) {
+        super.createBlockStateDefinition(pBuilder)
+        val connect_east = BooleanProperty.create("connect_east")
+        val connect_north = BooleanProperty.create("connect_north")
+        val connect_south = BooleanProperty.create("connect_south")
+        val connect_west = BooleanProperty.create("connect_west")
+        pBuilder.add(connect_east, connect_north, connect_south, connect_west)
+    }
+
+    fun neChang(level: Level, pos: BlockPos, neighbor: BlockPos) {
+        val connect_east = BooleanProperty.create("connect_east")
+        val connect_north = BooleanProperty.create("connect_north")
+        val connect_south = BooleanProperty.create("connect_south")
+        val connect_west = BooleanProperty.create("connect_west")
+
+
+
+        val blockEntity = level.getBlockEntity(neighbor ?: error("Neighbor position was null"))
+
+        println("Neighbor block entity: $blockEntity")
+        if(blockEntity != null) {
+
+
+            println("It exists")
+            // check if the blockentity is from this mod
+
+            if(blockEntity is CellTileEntity) {
+
+                println("It is a cell")
+
+                // get the direction of the neighbor from us
+
+                val direction = Direction.fromNormal(neighbor.subtract(pos))
+
+                println("FUCKING FUCKER KURWA FUCkering Direction: $direction")
+
+
+                if(direction?.equals(Direction.EAST) == true) {
+
+                    println("East")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_east, true))
+                }
+                if(direction?.equals(Direction.NORTH) == true) {
+                    println("North")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_north, true))
+                }
+                if(direction?.equals(Direction.SOUTH) == true) {
+                    println("South")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_south, true))
+                }
+                if(direction?.equals(Direction.WEST) == true) {
+                    println("West")
+                    level.setBlockAndUpdate(pos, level.getBlockState(pos).setValue(connect_west, true))
+                }
```

Comment:
Consider replacing with a `when` statement from Kotlin. It will clean up this logic a bit.